src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -370,74 +375,90 @@ else if (participation instanceof ProgrammingExerciseParticipation) {
     }
 
     /**
-     * DELETE /results/:id : delete the "id" result.
+     * DELETE /participations/:participationId/results/:resultId : delete the "id" result.
      *
+     * @param participationId the id of the participation to the result
      * @param resultId the id of the result to delete
      * @return the ResponseEntity with status 200 (OK)
      */
-    @DeleteMapping("/results/{resultId}")
+    @DeleteMapping("participations/{participationId}/results/{resultId}")
     @PreAuthorize("hasRole('TA')")
-    public ResponseEntity<Void> deleteResult(@PathVariable Long resultId) {
+    public ResponseEntity<Void> deleteResult(@PathVariable Long participationId, @PathVariable Long resultId) {
         log.debug("REST request to delete Result : {}", resultId);
-        Optional<Result> result = resultRepository.findById(resultId);
-        if (result.isPresent()) {
-            Participation participation = result.get().getParticipation();
-            Course course = participation.getExercise().getCourseViaExerciseGroupOrCourseMember();
-            if (!authCheckService.isAtLeastTeachingAssistantInCourse(course, null)) {
-                return forbidden();
-            }
-            resultRepository.deleteById(resultId);
-            return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, resultId.toString())).build();
+        Result result = resultRepository.findOneElseThrow(resultId);
+        Participation participation = result.getParticipation();
+        if (!participation.getId().equals(participationId)) {
+            badRequest("participationId", "400",
+                    "participationId of the path doesnt match the participationId of the participation corresponding to the result " + resultId + " !");
         }
-        return ResponseEntity.notFound().build();
+        Course course = participation.getExercise().getCourseViaExerciseGroupOrCourseMember();
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.TEACHING_ASSISTANT, course, null);
+        resultRepository.deleteById(resultId);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, resultId.toString())).build();
     }
 
     /**
-     * GET /results/submission/{submissionId} : get the result for a submission id
+     * GET /participations/:participationId/results/from-submission/:submissionId : get the result for a submission id
      *
+     * @param participationId the id of the participation to the submission
      * @param submissionId the id of the submission
      * @return the ResponseEntity with status 200 (OK) and the list of results in body
      */
-    @GetMapping(value = "/results/submission/{submissionId}")
+    @GetMapping("participations/{participationId}/results/from-submission/{submissionId}")
```

Comment:
```suggestion
    @GetMapping("participations/{participationId}/submissions/{submissionId}/results")
```

Reply:
Since some time a submission can have multiple results. I wonder how this affects this REST Call.
Can you check how this Rest call is used?

Reply:
Since some time a submission can have multiple results. I wonder how this affects this REST Call.
Can you check how this Rest call is used?

Reply:
This is the unused Endpoint from the client that should be removed, so it doesnt matter since this Endpoint gets removed anyways ^^